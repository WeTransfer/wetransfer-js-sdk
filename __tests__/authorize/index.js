const axios = require('axios');

const authorize = require('../../src/authorize');
const request = require('../../src/request');

jest.mock('axios');

describe('Authorize method', () => {
  beforeEach(() => {
    request.apiKey = 'secret-api-key';
    axios.mockImplementation(() =>
      Promise.resolve({ data: { token: 'json-web-token' } })
    );
  });

  afterEach(() => {
    axios.mockReset();
  });

  it('should create an authorize request', async () => {
    await authorize();
    expect(axios).toHaveBeenLastCalledWith({
      headers: {
        'Content-Type': 'application/json',
        'x-api-key': 'secret-api-key',
      },
      url: '/v2/authorize',
      data: null,
    });
  });

  it('should return a JWT token', async () => {
    const auth = await authorize();
    expect(auth).toEqual({
      token: 'json-web-token',
    });
  });

  it('should throw an error if request fails', async () => {
    try {
      axios.mockImplementation(() =>
        Promise.reject(new Error('Network error.'))
      );
      await authorize();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });
});
