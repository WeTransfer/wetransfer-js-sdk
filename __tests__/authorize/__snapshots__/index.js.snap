// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize method should throw an error if request fails 1`] = `[Error: The authorization call failed and no usable JWT could be found there. Please check your API Key.]`;
